modules/apigee_edge_teams/src/Form/TeamContextSwitcherForm.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge_teams\Form;
+
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a form for switching team.
+ */
+class TeamContextSwitcherForm extends FormBase implements ContainerInjectionInterface {
+
+  /**
+   * The current route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * TeamContextSwitcherForm constructor.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The current route match.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(RouteMatchInterface $route_match, EntityTypeManagerInterface $entity_type_manager) {
+    $this->routeMatch = $route_match;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('current_route_match'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'apigee_edge_teams_context_switcher_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, array $teams = []) {
+    $form_state->set('teams', $teams);
+
+    /** @var \Drupal\apigee_edge_teams\Entity\TeamInterface $team */
+    $options = [];
+    foreach ($teams as $id => $team) {
+      $options[$id] = $team->label();
```

Comment:
This list also should not display the current team (from the route) as an option because that does not make any sense.

Reply:
@mxr576 do teams have granular access control? How can I give permission to a user to create app in Team A and not Team B?

@mxr576 Also, I'd argue we need to keep the current team as an option. It gives the user context from navigating. Maybe the current team could be disabled in the select but it should be there.

Reply:
2. How do you disable an option in a select list? :) I mean, sure, you can display the current team somehow in the form, but not in the select list, otherwise the user could select it, click on "Go" which simply just reloads the page - maybe s/he is going to lose some form data because of that if s/he does that on a team related form. But this is just a small details compared with the who should see what question.

1. The possibility is there, but by default all team members shares the same settings, meaning that, if a team member can create apps then it can create team apps in all teams where s/he is a member. Check [apigee_edge_teams.api.php](https://github.com/apigee/apigee-edge-drupal/blob/8.x-1.x/modules/apigee_edge_teams/apigee_edge_teams.api.php#L63) and team member permission related classes. Even a user who is not member of a team can get access to that team related UIs via site-wide permissions or custom implements of `hook_apigee_edge_teams_developer_permissions_by_team_alter()` alter hook.

Reply:
Closing this since this is now outdated. We are now using a dropbutton instead of a form with a select.